consensus/blake3pow/consensus.go

```diff
@@ -399,19 +399,21 @@ func (blake3pow *Blake3pow) Prepare(chain consensus.ChainHeaderReader, header *t
 	return nil
 }
 
-// Finalize implements consensus.Engine, accumulating the block and uncle rewards,
-// setting the final state on the header
-func (blake3pow *Blake3pow) Finalize(chain consensus.ChainHeaderReader, header *types.Header, state *state.StateDB, txs []*types.Transaction, uncles []*types.Header) {
+// FinalizeAtContext implements consensus.Engine, accumulating the block and uncle rewards,
+// setting the final state on the header at an context.
+func (blake3pow *Blake3pow) FinalizeAtContext(chain consensus.ChainHeaderReader, header *types.Header, state *state.StateDB, txs []*types.Transaction, uncles []*types.Header, ctx int) {
```

Comment:
I don't love the idea of one chain computing the state root for a block in another chain. Everything else in our system asks the sub or dom when it needs data from sub or dom. We purposefully never try to compute sub or dom data. Is this strictly necessary?